patroni/ha.py

```diff
@@ -914,6 +914,13 @@ def _process_multisync_replication(self) -> None:
 
         # update synchronous standby list in dcs temporarily to point to common nodes in current and picked
         sync_common = voters & allow_promote
+
+        # Since synchronous strict mode requires at least one other replica, /sync should not be updated
+        # with an empty set
+        if global_config.is_synchronous_mode_strict and len(voters) != 0 and not sync_common:
```

Comment:
What about the following case?
We have a primary A and sync replica B.
At the same time when B is gone C is connected.
It will lead to the situation that `synchronous_standby_names` and `/sync` key will not be updated.
I don't think it is desired behavior.